### PR TITLE
parse: filter edge-case for podman-remote

### PR DESCRIPTION
**Pull request by flouthoc**
podman-remote has a bug where it sends `/` when no platform is set, patch is there to make sure older clients still work if backend is updated.

Closes: https://github.com/containers/buildah/issues/4763

#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Does this PR introduce a user-facing change?

```release-note
parse: filter edge-case for podman-remote
```

### CODE REVIEW

**Comment by rhatdan** [comment]
/approve
LGTM

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL

File: pkg/parse/parse.go [hunk 1]
```diff
@@ -518,7 +518,7 @@ func DefaultPlatform() string {
 // Platform separates the platform string into os, arch and variant,
 // accepting any of $arch, $os/$arch, or $os/$arch/$variant.
 func Platform(platform string) (os, arch, variant string, err error) {
-	if platform == "local" {
+	if platform == "local" || platform == "" || platform == "/" {
```
Comment (TomSweeneyRedHat) [inline pkg/parse/parse.go:521]: It might be nice to have a debug statement in here just to let people know the error triggered and the value found in the platform variable.

Comment (flouthoc) [reply pkg/parse/parse.go:521]: For these cases it will always return `default` without any errors but for any error case scenario it retruns error with the value used in `--platform` ( Line: 531 )

**Comment by flouthoc** [comment]
Rebased.

**Comment by edsantiago** [comment]
Unfortunate to have to special-case it here, but LGTM in principle. I have not actually run podman-buildah-treadmill tests against this.
